sycl/test/host-interop-task/host-task-two-queues.cpp

```diff
@@ -0,0 +1,74 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <vector>
+
+namespace S = cl::sycl;
+
+#define WIDTH 5
+#define HEIGHT 5
+
+void test() {
+  auto EH = [](S::exception_list EL) {
+    for (const std::exception_ptr &E : EL) {
+      throw E;
+    }
+  };
+
+  S::queue Q1(EH);
+  S::queue Q2(EH);
+
+  std::vector<int> DataA(WIDTH * HEIGHT, 2);
+  std::vector<int> DataB(WIDTH * HEIGHT, 3);
+  std::vector<int> DataC(WIDTH * HEIGHT, 1);
+
+  S::buffer<int, 2> BufA{DataA.data(), S::range<2>{WIDTH, HEIGHT}};
+  S::buffer<int, 2> BufB{DataB.data(), S::range<2>{WIDTH, HEIGHT}};
+  S::buffer<int, 2> BufC{DataC.data(), S::range<2>{WIDTH, HEIGHT}};
+
+  auto CG1 = [&](S::handler &CGH) {
+    auto AccA = BufA.get_access<S::access::mode::read>(CGH);
+    auto AccB = BufB.get_access<S::access::mode::read>(CGH);
+    auto AccC = BufC.get_access<S::access::mode::read_write>(CGH);
+    auto Kernel = [=](S::nd_item<2> Item) {
+      size_t W = Item.get_global_id(0);
+      size_t H = Item.get_global_id(1);
+      AccC[W][H] += AccA[W][H] * AccB[W][H];
+    };
+    CGH.parallel_for<class K1>(S::nd_range<2>({WIDTH, HEIGHT}, {1, 1}), Kernel);
+  };
+
+  auto CG2 = [&](S::handler &CGH) {
+    auto AccA = BufA.get_access<sycl::access::mode::read>(CGH);
+    auto AccB = BufB.get_access<sycl::access::mode::read>(CGH);
+    auto AccC = BufC.get_access<sycl::access::mode::read_write>(CGH);
+
+    CGH.codeplay_host_task([=] {
+      for (size_t I = 0; I < WIDTH; ++I)
+        for (size_t J = 0; J < HEIGHT; ++J) {
+          std::cout << "C[" << I << "][" << J << "] = " << AccC[I][J]
+                    << std::endl;
+        }
+    });
+  };
+
+  Q1.submit(CG1);
+  Q2.submit(CG2);
+  Q2.submit(CG1);
+  Q1.submit(CG2);
+
+  Q1.wait_and_throw();
+  Q2.wait_and_throw();
+
+  for (size_t I = 0; I < WIDTH; ++I)
+    for (size_t J = 0; J < HEIGHT; ++J)
+      assert(DataC[I * HEIGHT + J] == (1 + 2 * 3) + 2 * 3);
+}
+
+int main(void) {
+  test();
+  return 0;
+}
```

Comment:
This variant of the test above:

```cpp
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
// RUN: %CPU_RUN_PLACEHOLDER %t.out
// RUN: %GPU_RUN_PLACEHOLDER %t.out
// RUN: %ACC_RUN_PLACEHOLDER %t.out

#include <CL/sycl.hpp>
#include <vector>

namespace S = cl::sycl;

#define WIDTH 5
#define HEIGHT 5

void test() {
  auto EH = [](S::exception_list EL) {
    for (const std::exception_ptr &E : EL) {
      throw E;
    }
  };

  S::queue Q1(EH);
  S::queue Q2(EH);

  std::vector<int> DataA(WIDTH, 2);
  std::vector<int> DataB(WIDTH, 3);
  std::vector<int> DataC(WIDTH, 1);

  S::buffer<int, 1> BufA{DataA.data(), S::range<1>{WIDTH}};
  S::buffer<int, 1> BufB{DataB.data(), S::range<1>{WIDTH}};
  S::buffer<int, 1> BufC{DataC.data(), S::range<1>{WIDTH}};

  auto CG1 = [&](S::handler &CGH) {
    auto AccA = BufA.get_access<S::access::mode::read>(CGH);
    auto AccB = BufB.get_access<S::access::mode::read>(CGH);
    auto AccC = BufC.get_access<S::access::mode::read_write>(CGH);
    auto Kernel = [=](S::nd_item<1> Item) {
      size_t W = Item.get_global_id(0);
      AccC[W] += AccA[W] * AccB[W];
    };
    CGH.parallel_for<class K1>(S::nd_range<1>(WIDTH, 1), Kernel);
  };

  auto CG2 = [&](S::handler &CGH) {
    auto AccA = BufA.get_access<sycl::access::mode::read>(CGH);
    auto AccB = BufB.get_access<sycl::access::mode::read>(CGH);
    auto AccC = BufC.get_access<sycl::access::mode::read_write>(CGH);

    CGH.codeplay_host_task([=] {
      for (size_t I = 0; I < WIDTH; ++I)
          std::cout << "C[" << I << "]" << AccC[I] << std::endl;
    });
  };

  size_t NTIMES = 100;
  for (int i = 0; i < NTIMES; i++) {
    Q1.submit(CG1);
    Q2.submit(CG2);
    Q2.submit(CG1);
    Q1.submit(CG2);
  }

  Q1.wait_and_throw();
  Q2.wait_and_throw();

  for (size_t I = 0; I < WIDTH; ++I)
      assert(DataC[I] == (1  + 2*3 * NTIMES * 2));
}

int main(void) {
  test();
  return 0;
}
```

triggers  an assertion:

```

# command stderr:
host-task-two-queues.cpp.tmp.out: /opt/build_dirs/llvm-codeplay/sycl/source/detail/scheduler/commands.cpp:192: static
cl::sycl::detail::EmptyCommand* cl::sycl::detail::DispatchHostTask::findUserEmptyCommand(cl::sycl::detail::Command*):
Assertion `ThisCmd->MUsers.size() == 1 && "Only a single user is expected for host task command"' failed.

error: command failed with exit status: -6
```

Reply:
Done. Fixed. Updated the test.

Reply:
@Ruyk Done. Fixed. Updated the test.